actionpack/lib/action_dispatch/middleware/debug_exceptions.rb

```diff
@@ -142,7 +142,7 @@ def log_error(request, wrapper)
         message = []
         message << "  "
         message << "#{exception.class} (#{exception.message}):"
-        message.concat(exception.annotated_source_code) if exception.respond_to?(:annotated_source_code)
+        message.concat(exception.annotated_source_code || []) if exception.respond_to?(:annotated_source_code)
```

Comment:
Since https://github.com/rails/rails/pull/36532, it's no longer possible for `annotated_source_code` to return nil.

I think we should just delete the test.

Reply:
Okay. I will remove the NilAnnotedSourceCodeError test.